### PR TITLE
correct excel mime types

### PR DESCRIPTION
**Pull request by pjfanning**
https://github.com/OpenRefine/OpenRefine/issues/2595

### CODE REVIEW

File: main/webapp/modules/core/MOD-INF/controller.js [hunk 1]
```diff
@@ -223,6 +223,8 @@ function registerImporting() {
 
   IM.registerFormat("text/xml", "XML files", "XmlParserUI", new Packages.com.google.refine.importers.XmlImporter());
   IM.registerFormat("binary/text/xml/xls/xlsx", "Excel files", "ExcelParserUI", new Packages.com.google.refine.importers.ExcelImporter());
+  IM.registerFormat("application/vnd.ms-excel", "Excel files", "ExcelParserUI", new Packages.com.google.refine.importers.ExcelImporter());
+  IM.registerFormat("application/vnd.openxmlformats-officedocument.spreadsheetml.*", "Excel files", "ExcelParserUI", new Packages.com.google.refine.importers.ExcelImporter());
```
Comment (wetneb) [inline main/webapp/modules/core/MOD-INF/controller.js:227]: This should be removed - we should only need to add a mapping from the MIME type to the internal format identifier, with `IM.registerMimeType` (see below).

File: main/webapp/modules/core/MOD-INF/controller.js [hunk 2]
```diff
@@ -245,8 +247,8 @@ function registerImporting() {
   IM.registerExtension(".json", "text/json");
   IM.registerExtension(".js", "text/json");
 
-  IM.registerExtension(".xls", "binary/text/xml/xls/xlsx");
-  IM.registerExtension(".xlsx", "binary/text/xml/xls/xlsx");
+  IM.registerExtension(".xls", "application/vnd.ms-excel");
+  IM.registerExtension(".xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet");
```
Comment (wetneb) [inline main/webapp/modules/core/MOD-INF/controller.js:251]: This should be kept: here, "binary/text/xml/xls/xlsx" serves as an internal identifier for the format, and does not need to be an official MIME type. The correct place to add the mapping is further down, with `IM.registerMimeType`.

File: main/webapp/modules/core/MOD-INF/controller.js [hunk 3]
```diff
@@ -291,7 +291,8 @@ function registerImporting() {
   IM.registerMimeType("application/x-excel", "binary/text/xml/xls/xlsx");
   IM.registerMimeType("application/xls", "binary/text/xml/xls/xlsx");
   IM.registerMimeType("application/x-xls", "binary/text/xml/xls/xlsx");
-  
+  IM.registerMimeType("application/vnd.openxmlformats-officedocument.spreadsheetml.*", "binary/text/xml/xls/xlsx");
```
Comment (pjfanning) [inline main/webapp/modules/core/MOD-INF/controller.js:294]: * I'm not sure if wildcards work here
* application/vnd.ms-excel already has an entry

Comment (wetneb) [reply main/webapp/modules/core/MOD-INF/controller.js:294]: good catch, wildcards are indeed not supported yet, sadly. Perhaps it is still worth keeping here for reference, it does not hurt: I'll add an issue to add support for wildcards which would enable it. Unless there is a small number of possible values for this wildcard, in which case we could add all the variants directly?

Comment (wetneb) [reply main/webapp/modules/core/MOD-INF/controller.js:294]: If we keep this line for now, we should add a comment about the fact that the wildcard is not supported yet: https://github.com/OpenRefine/OpenRefine/issues/2598

**Comment by pjfanning** [comment]
I removed the wildcard

**Comment by thadguidry** [comment]
@pjfanning Thank you so much for putting a few hours into helping us out with all of this! I was blown away waking up this morning and seeing all your activity!

**Pull request merged**